### PR TITLE
fix #8385 feat(cirrus): Set up test cases and test coverage

### LINKED ISSUE #8385: Set up support fot test cases
Set up support for python test and coverage for cirrus service

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/EXP-3214)

### PR DESCRIPTION
**Pull request by yashikakhurana**
Because

- We added Cirrus FastAPI recently

This commit

- Adds a test support- pytest, coverage and make command to run cirrus test cases
- You can run the command `make cirrus_test` to run cirrus test cases
image
closes #8385

### CODE REVIEW

File: cirrus/server/Dockerfile [hunk 1]
```diff
@@ -3,4 +3,5 @@ WORKDIR /usr/src/cirrus
 COPY cirrus/server/requirements.txt ./
 RUN pip install --no-cache-dir -r requirements.txt
 COPY . .
-CMD ["uvicorn", "cirrus.main:app", "--host", "0.0.0.0", "--port", "8001"]
\ No newline at end of file
+CMD ["uvicorn", "cirrus.main:app", "--host", "0.0.0.0", "--port", "8001"]
+
```
Comment (jrbenny35) [inline cirrus/server/Dockerfile:7]: Looks like an extra space was added here

File: cirrus/server/requirements.txt [hunk 2]
```diff
@@ -1,2 +1,7 @@
 fastapi==0.92.0
-uvicorn==0.20.0
\ No newline at end of file
+uvicorn==0.20.0
+
+#tests
+pytest==7.2.2
+httpx==0.23.3
+pytest-cov==4.0.0
```
Comment (jrbenny35) [inline cirrus/server/requirements.txt]: Is there a reason you are not using hashes here?

File: cirrus/server/tests/conftest.py [hunk 3]
```diff
@@ -0,0 +1,9 @@
+from pytest import fixture
+from fastapi.testclient import TestClient
+from cirrus.main import app
+
+
+@fixture(scope="module")
+def client():
+    with TestClient(app) as c:
+        yield c
```
Comment (jrbenny35) [inline cirrus/server/tests/conftest.py:9]: Curious to know why you are using a yield here

**Review by jrbenny35 — CHANGES_REQUESTED** [review]
Are you planning to add a circleci test job?

**Comment by yashikakhurana** [comment]
> Are you planning to add a circleci test job?

yes that will be added in separate PR

File: cirrus/server/tests/conftest.py (hunk 3 above)
Comment (yashikakhurana) [reply cirrus/server/tests/conftest.py:9]: oh, the reason is that the yield statement returns the client object to any functions that call it. Once pytest finishes running all the tests, it also automatically closes the TestClient context/cleanups resource when it's ready to exit from this module.

File: cirrus/server/requirements.txt (hunk 2 above)
Comment (yashikakhurana) [reply cirrus/server/requirements.txt]: To be honest I don't have a strong reason why I haven't used hashes, that's only my personal choice I guess , but would love to hear your thoughts :)

File: cirrus/server/Dockerfile (hunk 1 above)
Comment (yashikakhurana) [reply cirrus/server/Dockerfile]: my bad, fixed

**Review by jeddai — APPROVED** [review]
Looks good, excited to see this plugged into ci soon!

**Pull request merged**